editor/components/block-list/index.js

```diff
@@ -62,15 +70,45 @@ class BlockList extends Component {
 		document.addEventListener( 'copy', this.onCopy );
 		document.addEventListener( 'cut', this.onCut );
 		window.addEventListener( 'mousemove', this.setLastClientY );
+		// Needs document to also work in inspector.
+		// In other words, if there is selection, but no focus.
+		document.addEventListener( 'keydown', this.onKeyDown );
 	}
 
 	componentWillUnmount() {
 		document.removeEventListener( 'copy', this.onCopy );
 		document.removeEventListener( 'cut', this.onCut );
 		window.removeEventListener( 'mousemove', this.setLastClientY );
+		document.removeEventListener( 'keydown', this.onKeyDown );
 	}
 
 	componentWillReceiveProps( nextProps ) {
+		const prevCommonName = this.commonName;
```

Comment:
Could we consider moving the logic for handling the shortcuts to a different component? This doesn't seem particularly relevant to the block listing, and would be easy enough to transport into a separate (non-visual) component which is rendered as a child of the `BlockList` root return value.

Reply:
I will, if the general direction of this PR seems good.